### PR TITLE
 Refactor schema

### PR DESCRIPTION
**Pull request by fearthecowboy**
- pulled in linq project to make perf changes 
  - `values`, `keys`, `items` return straight Iterable<> for faster use
  - **important** -- `items()` now returns an array [key,value] instead of an object {key, value} -- change usage from `for( const {key,value} of items(foo) (` to  `for ( const [key, value] of items(foo) ) `
  - switched to using `Object.entries()` and `Object.values()` for object/dictionaries 
  - introduced `linq.values`, `linq.keys`, `linq.items` when you want a linq operation on the iterable (ie, returns a `IterableWithLinq<>`
- schemas are now handled with iterators, no direct pushing to the target collection
- aliases are created when there is a description/nullable on a primitive type
-

### CODE REVIEW

File: adl/core/serialization/openapi/v3/components.ts [hunk 1]
```diff
@@ -1,52 +1,84 @@
 import { v3 } from '@azure-tools/openapi';
+import { Alias as A } from '../../../model/alias';
```
Comment (nguerrera) [inline adl/core/serialization/openapi/v3/components.ts:2]: Maybe a more descriptive disambiguating name than 'A'?

File: adl/core/test/scenarios/v3/single/output/github.attic.yaml [hunk 2]
```diff
@@ -1 +1,146 @@
-{}
+paths:
```
Comment (nguerrera) [inline adl/core/test/scenarios/v3/single/output/github.attic.yaml:1]: stuff is going *into* the attic?

Comment (fearthecowboy) [reply adl/core/test/scenarios/v3/single/output/github.attic.yaml:1]: WTF  it doesn't look like that locally. 

One sec.

Comment (fearthecowboy) [reply adl/core/test/scenarios/v3/single/output/github.attic.yaml]: Ok, I'm confused -- I pushed it, and it was correct in my tree, but the PR didn't pick it up for a while. I've never seen that happen

**Pull request closed without merging**